### ISSUE TITLE
Connection re-use does not happen in some cf commands

### ISSUE THREAD

**Issue by akskap** [cloudfoundry/cli#1329]
## Command  
```
cf service-access -b <broker_name>
```

## What occurred

auth request failed: Error performing request: Post https://login.cf.stagingazure.xxxxx.com/oauth/token: dial tcp 13.81.xxx.xx:443: i/o timeout
TIP: If you are behind a firewall and require an HTTP proxy, verify the https_proxy environment variable is correctly set. Else, check your network connection.

Is it expected that the above command does not re-use existing connections. However, I did notice that something like `cf orgs` does. Our current setup has 542 orgs and the above commands seems to close TCP connection on every subsequent request

```
REQUEST: [2018-02-18T07:08:46Z]
GET /v2/service_brokers?q=name%3Aservice-fabrik-broker HTTP/1.1^M
Host: api.cf.stagingazure.hanavlab.ondemand.com^M
Accept: application/json^M
Authorization: [PRIVATE DATA HIDDEN]
Connection: close^M
Content-Type: application/json^M
User-Agent: go-cli 6.34.1+bbdf81482.2018-01-17 / linux^M
^M


RESPONSE: [2018-02-18T07:08:46Z]
HTTP/1.1 200 OK^M
Connection: close
Content-Length: 552^M
Content-Type: application/json;charset=utf-8^M

.........and so on
```

## What you expected to occur

Command to pass without error

## CLI Version  

cf version 6.34.1+bbdf81482.2018-01-17

## CC API Endpoint Version  

api endpoint:   https://api.cf.stagingazure.xxx.com
api version:    2.101.0

## CF Trace

(Please find below a trimmed down version of CF_TRACE output)
```
REQUEST: [2018-02-18T07:08:46Z]
POST /oauth/token HTTP/1.1
Host: login.cf.stagingazure.xxx.com
Accept: application/json
--
RESPONSE: [2018-02-18T07:08:46Z]
HTTP/1.1 200 OK
Connection: close
Transfer-Encoding: chunked
--
REQUEST: [2018-02-18T07:08:46Z]
GET /v2/service_brokers?q=name%3Aservice-fabrik-broker HTTP/1.1
Host: api.cf.stagingazure.xxx.com
Accept: application/json
--
RESPONSE: [2018-02-18T07:08:46Z]
HTTP/1.1 200 OK
Connection: close
Content-Length: 552
--
REQUEST: [2018-02-18T07:08:46Z]
GET /v2/services?q=service_broker_guid%3A73062aa6-9cd7-48fd-8d7f-5385904d0051 HTTP/1.1
Host: api.cf.stagingazure.xxx.com
Accept: application/json
--
RESPONSE: [2018-02-18T07:08:46Z]
HTTP/1.1 200 OK
Connection: close
Content-Length: 8282
--
REQUEST: [2018-02-18T07:08:46Z]
GET /v2/service_plans?q=service_guid+IN+2302e76f-4186-43d5-8a47-b8594939c9d4%2Cae3e86df-f60d-45ba-9ed9-a09d7814c493%2Cce6a0192-bf07-481d-adfb-710a1663781c%2Cfc36bcae-141c-4ed7-b8dd-b384d9b5d996%2C58ebdac0-276b-42dc-90df-711b30283f1e%2C6dd
4c56c-cb11-434f-a660-292fa78e8799 HTTP/1.1
Host: api.cf.stagingazure.xxx.com
Accept: application/json
--
RESPONSE: [2018-02-18T07:08:46Z]
HTTP/1.1 200 OK
Connection: close
Content-Length: 76692
--
REQUEST: [2018-02-18T07:08:46Z]
GET /v2/service_plans?order-direction=asc&page=2&q=service_guid%20IN%202302e76f-4186-43d5-8a47-b8594939c9d4,ae3e86df-f60d-45ba-9ed9-a09d7814c493,ce6a0192-bf07-481d-adfb-710a1663781c,fc36bcae-141c-4ed7-b8dd-b384d9b5d996,58ebdac0-276b-42dc-
90df-711b30283f1e,6dd4c56c-cb11-434f-a660-292fa78e8799&results-per-page=50 HTTP/1.1
Host: api.cf.stagingazure.xxx.com
Accept: application/json

... and so on
```

## Platform & Shell Details  

Ubuntu 16.04.3 LTS

## Any other relevant information  

Concourse VM from which this command is executed, is behind [Azure SNAT](https://blogs.msdn.microsoft.com/mast/2015/07/13/azure-snat/), which by default exposes 160 ports as a soft limit. So, current understanding is that with large number of CF orgs, commands like `cf service-access -b <broker_name>` result in SNAT port exhaustion and the script which includes similar commands fail.

**Comment by jbpivotal** [comment]
We'll refactor this command later, and that will resolve this issue. Closing.

**Comment by akskap** [comment]
@jbpivotal Thanks for your comments. Can you please let me know if not re-using connections was intentional for such commands? 

Asking because this is currently blocking us and we might have to work on an external utility that interacts directly with the CC with connection re-use

**Comment by akskap** [comment]
@jbpivotal Possible for you to give a rough estimate for the availability of the refactored commands ?

**Comment by jbpivotal** [comment]
@akskap Thanks for letting me know this issue is blocking you. I've moved it up higher in priority so we'll likely pick it up in the next week or two. You can view the story here: https://www.pivotaltracker.com/story/show/126256853

**Comment by jbpivotal** [comment]
@akskap We spent some time investigating a refactor of the `serivce-access` command, as we believe that will resolve your issue. Unfortunately, it looks like a decent amount of work that would fully occupy a pair of engineers for at least 3 weeks or so. (We estimate based on complexity, not time, and we identified a 5-point story, a 3-point story, and another not-yet-pointed story.) As a result, I'm keeping this work in the backlog but moving it below some other features that we believe have wider demand relative to the development time cost.
